MetaMask/metamask-mobile
# feat: Add delay for android browser permission dialog

**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

When a webpage in the MM mobile browser requests the `camera` permission, a prompt is shown for users to accept or deny this request. However, in cases where a user is browsing very quickly, it is possible for a user to accidentally accept the permission without realizing it. 

To resolve this, we are introducing a 0.5s delay where the `allow` button is not clickable. After this delay, the permission can be accepted as needed.

<img width="384" alt="Screenshot 2023-07-05 at 3 55 58 PM" src="https://github.com/MetaMask/metamask-mobile/assets/15018469/519b8e46-f986-4bd0-affb-e7ec0fadb110">


**Manual Testing Steps**

* Run this branch in the android emulator
* Visit https://webcamtests.com/ 
* Click `Test my cam` and immediately click `allow` on the permission prompt
    * You can refresh the page to try again if you aren't quick enough
    * If you have not previously approved the OS Level camera permission, accept that prompt and refresh the page to try agian. 
* You will see that the allow button is not responsive for 500ms

https://github.com/MetaMask/metamask-mobile/assets/15018469/7ccc0e86-55f5-42a8-8582-b4bf6f4102ed


**Checklist**
